### PR TITLE
Fix stack

### PR DESCRIPTION
**Pull request by jeanlaroche**
Add a ValueError for an empty input list

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
That's a lot of commits :-) you probably need to rebase against master or something :-)

File: pythran/pythonic/numpy/stack.hpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include <pythonic/include/numpy/stack.hpp>
 #include <pythonic/numpy/concatenate.hpp>
+#include "pythonic/builtins/len.hpp"
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/stack.hpp]: Also need to include the exception header

File: pythran/pythonic/numpy/stack.hpp [hunk 2]
```diff
@@ -14,6 +15,9 @@ namespace numpy
                  types::array<long, ArraySequence::value_type::value + 1>>
   stack(ArraySequence const &args, long axis)
   {
+    if(builtins::len(args) == 0)
+        throw pythonic::types::ValueError("need at least one array to stack");
+
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/stack.hpp]: Add a test case? To test that an exception is raised, unittest provides `assertRaises`, e.g. in `pythran/tests/test_conversion.py`

Comment (jeanlaroche) [reply pythran/pythonic/numpy/stack.hpp]: Done, but I'm not sure whether it's done right... I'm not sure how to verify that the test actually tests something and succeeds! (I get a pass, but I'm not 100% sure about all this)

File: pythran/pythonic/numpy/stack.hpp [hunk 3]
```diff
@@ -34,7 +38,8 @@ namespace numpy
     // Create a new empty list.
     types::list<types::ndarray<
         typename ArraySequence::value_type::dtype,
-        types::array<long, ArraySequence::value_type::value + 1>>> bi(0);
+        types::array<long, ArraySequence::value_type::value + 1>>>
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/stack.hpp]: useless change?

File: pythran/pythonic/numpy/stack.hpp [hunk 4]
```diff
@@ -51,7 +56,8 @@ namespace numpy
         types::ndarray<
             typename details::stack_helper_t<Tys...>::dtype,
             types::array<long, details::stack_helper_t<Tys...>::value>>,
-        sizeof...(Tys)> vargs{{std::get<Is>(args)...}};
+        sizeof...(Tys)>
+        vargs{{std::get<Is>(args)...}};
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/stack.hpp]: useless change?

File: pythran/tests/test_numpy_func2.py [hunk 5]
```diff
@@ -315,6 +315,11 @@ def test_concatenate2(self):
     def test_concatenate3(self):
         self.run_test("def np_concatenate3(a): from numpy import array, concatenate ; return concatenate([[1],a + a])", numpy.array([1, 2]), np_concatenate3=[NDArray[int,:]])
 
+    def test_hstack_empty(self):
+        code = 'def np_test_stack_empty(a): import numpy as np;return np.stack(())'
+        with self.assertRaises(BaseException):
```
Comment (serge-sans-paille) [inline pythran/tests/test_numpy_func2.py]: Perfect! Could be `ValueError` instead though.

File: pythran/pythonic/numpy/stack.hpp (hunk 3 above)
Comment (jeanlaroche) [reply pythran/pythonic/numpy/stack.hpp]: That's clang-format

**Comment by jeanlaroche** [comment]
The unit test was not working right because pythran was detecting the empty input. It's working correctly now.

File: pythran/pythonic/numpy/stack.hpp (hunk 3 above)
Comment (serge-sans-paille) [reply pythran/pythonic/numpy/stack.hpp]: Unfortunately not, see the validation... you need to run `clang-format-3.6`, sorry about that

File: pythran/pythonic/numpy/stack.hpp [hunk 6]
```diff
@@ -1,6 +1,8 @@
 #ifndef PYTHONIC_NUMPY_STACK_HPP
 #define PYTHONIC_NUMPY_STACK_HPP
 
+#include "pythonic/builtins/len.hpp"
+#include "pythonic/types/exceptions.hpp"
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/stack.hpp]:
```
#include "pythonic/builtins/ValueError.hpp"
```

would be better.

**Comment by serge-sans-paille** [comment]
LGTM, can you rebase againstmaster and squash the two commits?Thanks!

**Pull request merged**